test/Npgsql.Tests/Types/JsonDynamicTests.cs

```diff
@@ -235,23 +236,19 @@ await AssertType<WeatherForecast>(
                 Summary = "Partly cloudy",
                 TemperatureC = 10
             },
-            """{"$type":"extended","TemperatureF":49,"Date":"2019-09-01T00:00:00","TemperatureC":10,"Summary":"Partly cloudy"}""",
+            sql,
             PostgresType,
             NpgsqlDbType,
-            isDefaultForReading: false,
+            isDefault: false,
```

Comment:
Any reason to not check default for writing? It should work for `jsonb` just like it already works for `json` (you need to update `EnableDynamicJson` above to pass WeatherForecast in the `jsonbClrTypes` parameter.

Reply:
Truthfully, I don't understand the deep depths of what these test parameters do, only that the tests above use a similar syntax and without this change it errors on a 4th parameter saying it expects "jsonb" but receives "json".

Reply:
@onionhammer sure, that's totally fine. AssertType simple calls AssertTypeWrite and AssertTypeRead; the former sends the .NET type to PostgreSQL, reads back the result as a `text` and asserts on the result, while AssertType latter tells PG to convert the string representation to the type (jsonb in this case), and reads the result as the .NET type.

Reply:
BTW if you're feeling lost on the tests, don't hesitate to drop a note - I can do the extra work here and push a commit to your PR.

Reply:
> BTW if you're feeling lost on the tests, don't hesitate to drop a note - I can do the extra work here and push a commit to your PR.

@rojiThank you for offering! I would appreciate that greatly if you've got the bandwidth!

Reply:
@onionhammer no worries. You can leave things wherever you end up and I'll pick up the work from there tomorrow.